### ISSUE TITLE
centering text with anchor set to middle dose not work

### ISSUE THREAD

**Issue by rdfriedl** [svgdotjs/svg.js#445]
i came across this the other day, not sure if its a bug or not, but it does not seem like behavior that you would expect.

if you set a text elements attr "text-anchor" to "middle" or "end" it fails to center on the x axis.
[example](https://jsfiddle.net/bek44jrp/2/)

thoughts?

**Comment by Fuzzyma** [comment]
The positioning of the text is anchor-independent. I agree that this is confusing and I never thought about that.
You have to know, that you are always moving the anchor of the text which is not necessarily the left edge. However the lib assumes that.
So when you center the text with anchor, it is centered around the *would be*-left edge.

I have to admit, that I wouldnt know how to handle all the special cases.
In my opinion `move` should always move the anchor point. That said, its hard to determine a logical behavior of the center function. I am open for any suggestions!

Quickfix: dont use center and anchor together

**Comment by rdfriedl** [comment]
i think as it stands right now, the <code>move()</code> function moves the text based on its anchor.
though as you said the <code>center()</code> function moves the text based on the left side, of course this only works if the anchor is on the left.

i created a small function in my project to fix this
```javascript
SVG.extend(SVG.Text, {
	cx: function(v){
		switch(this.attr('text-anchor')){
			default:
			case 'start':
				this.x(v - this.bbox().width/2);
				break;
			case 'middle':
				this.x(v);
				break;
			case 'end':
				this.x(v + this.bbox().width/2);
				break;
		}
		return this;
	}
});
```
though i doubt its something that would be worth putting into the library, it seems a little messing and kinda big to fix such a small bug.

im not sure how the center function should work on text, buti thought id bring this up since it did seem like a odd behavior.

**Comment by Fuzzyma** [comment]
At the moment the center method works like for any other shape with the left edge.
Thats the only reason why it is as it is.

However I guess your solution makes sense in some way.
I will let this open for discussion

**Comment by Fuzzyma** [comment]
will be fixed with #787 so I will close this one
